src/imperative/imperative_utils.h

```diff
@@ -60,6 +60,9 @@ struct EngineOprSeg {
 
 using MemoryPlanVector = std::vector<MemoryPlanInfo>;
 
+// cached segment operator name (needs a longer lifecycle than cached_seg_opr_)
+static std::unordered_set<std::string> cached_seg_opr_names_;
```

Comment:
Could this be a potential memory hug in case a lot of models are instantiated and disposed within the same application lifecycle? Or when does the clean up happen?

Reply:
Valid question. I don't see any explicit memory cleanup so I'm guessing it will be cleaned up when it goes out of scope? @joapolarbear you could either 
a. justify why we don't need to explicitly clean up this set 
or better
b. clean it once it is not needed